### PR TITLE
Implement updating and deleting DID based identities.

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

File: lib/services.identity.js [hunk 1]
```diff
@@ -596,7 +613,19 @@ function resolveIdentitySlug(name, options, callback) {
...
   async.auto({
-    getId: getIdentityIdFromUrl.bind(null, req),
+    getId: function(callback) {
+      if(req.params.identity && operation.id) {
+        return callback(new BedrockError(
+          'The identity could not be updated.', 'UpdateIdentityFailed', {
+          httpStatusCode: 400,
+          public: true
+        }));
+      }
```
Comment (mattcollier) [inline lib/services.identity.js]: How's this error?

Comment (dlongley) [inline lib/services.identity.js]: Eh, it's unclear what the problem is. It should probably be a validation error. At the very least we should say `'The identity could not be updated due to invalid input; the`id`property must not be provided.'`.

The real fix would probably be something like running a real quick `validate.validateInstance(<on-the-fly schema here>, ...)` check that just makes sure `id` is not in the operation and then return the error from that if there is one.

File: lib/services.identity.js [hunk 2]
```diff
@@ -407,6 +407,19 @@ bedrock.events.on('bedrock-express.configure.routes', function(app) {
     });
   });
 
+  app.delete(basePath,
+    ensureAuthenticated,
+    validate({query: 'services.identity.deleteIdentityParams'}),
+    function(req, res, next) {
+    brIdentity.setStatus(
```
Comment (dlongley) [inline lib/services.identity.js]: Fix indent.

**Comment by dlongley** [comment]
Just one comment to address. Then, if this passes testing, LGTM, merge.

**Pull request merged**